aws/sdk/examples/sts/src/bin/credentials-provider.rs

```diff
@@ -19,12 +18,7 @@ async fn main() -> Result<(), dynamodb::Error> {
     // credentials cached. See the docs on the builder for the various configuration options,
     // such as timeouts, default expiration times, and more.
     let sts_provider = LazyCachingCredentialsProvider::builder()
-        // `LazyCachingCredentialsProvider` requires an implementation of async sleep so that
-        // it can be used with different async runtimes. The sleep function will be used for timeouts.
-        // Since we're using Tokio in this example, we use the `TokioSleep` implementation, which
-        // requires the `timeout-tokio` feature in the `aws-auth` dependency.
-        .sleeper(TokioSleep::new())
-        .loader(async_provide_credentials_fn(move || {
+        .load(async_provide_credentials_fn(move || {
             let client = client.clone();
```

Comment:
If taking that approach, it's probably better to implement `AsyncProvideCredentials` on a struct instead, because you need a way to bind the STS `Client` to the function.